pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -91,7 +91,15 @@ func (a Adapter) Create() (err error) {
 	}
 
 	deploymentSpec := kclient.GenerateDeploymentSpec(*podTemplateSpec)
-
+	var containerPorts []corev1.ContainerPort
+	for _, c := range deploymentSpec.Template.Spec.Containers {
+		if len(containerPorts) == 0 {
+			containerPorts = c.Ports
+		} else {
+			containerPorts = append(containerPorts, c.Ports...)
+		}
+	}
```

Comment:
what happens if multiple components have the same endpoint, are we appending the same thing twice here? What happens when that is passed to `GenerateServiceSpec`

Reply:
There cannot be multiple components have the same endpoint, see Tomas's comment: 
```
all the dockerimage components are containers in one pod, and they can't expose same port.
```

Reply:
oh ok thx, We should see if push parse validation is doing it for the endpoints or else add it in 🤔

Reply:
Tomas has already tested it: 
```
▶ odo push
 •  Push devfile component spring-boot-http-booster  ...
 ✗  Failed to start component with name spring-boot-http-booster.
Error: Failed to start the component: unable to update Service for spring-boot-http-booster: Service "spring-boot-http-booster" is invalid: [spec.ports[1].name: Duplicate value: "8080-tcp", spec.ports[1]: Duplicate value: core.ServicePort{Name:"", Protocol:"TCP", Port:8080, TargetPort:intstr.IntOrString{Type:0, IntVal:0, StrVal:""}, NodePort:0}]
```